app/src/main/java/com/capstone/event_finder/network/EventDatabase.java

```diff
@@ -0,0 +1,56 @@
+package com.capstone.event_finder.network;
+
+import android.content.Context;
+import android.os.AsyncTask;
+
+import androidx.annotation.NonNull;
+import androidx.room.Database;
+import androidx.room.Room;
+import androidx.room.RoomDatabase;
+import androidx.sqlite.db.SupportSQLiteDatabase;
+
+import com.capstone.event_finder.interfaces.EventDao;
+import com.capstone.event_finder.models.Event;
+
+@Database(entities = {Event.class}, version = 1)
+public abstract class EventDatabase extends RoomDatabase {
+    private static volatile EventDatabase instance;
+    private static final RoomDatabase.Callback roomCallback = new RoomDatabase.Callback() {
+        @Override
+        public void onOpen(@NonNull SupportSQLiteDatabase db) {
+            super.onOpen(db);
+            new populateDbAsyncTask(instance);
+        }
+    };
+
+    public static synchronized EventDatabase getInstance(Context context) {
+        if (instance == null) {
+            synchronized (EventDatabase.class) {
+                if (instance == null) {
+                    instance = Room.databaseBuilder(context.getApplicationContext(),
+                                    EventDatabase.class, "event_database")
+                            .fallbackToDestructiveMigration()
```

Comment:
Since we discussed Room versions and migrating old database data schemas (like the zip code in US vs. UK example), I think fallbackToDestructiveMigration would enable Room to destroy and recreate the database in the different version. The word fallback likely implies that this happens only if the migration is not as successful as it should be so it's necessary to destroy and recreate. Would love to discuss this further in our next 1:1!